client/allocrunner/networking_cni.go

```diff
@@ -226,7 +228,101 @@ func (c *cniNetworkConfigurator) Teardown(ctx context.Context, alloc *structs.Al
 		return err
 	}
 
-	return c.cni.Remove(ctx, alloc.ID, spec.Path, cni.WithCapabilityPortMap(getPortMapping(alloc, c.ignorePortMappingHostIP)))
+	if err := c.cni.Remove(ctx, alloc.ID, spec.Path, cni.WithCapabilityPortMap(getPortMapping(alloc, c.ignorePortMappingHostIP))); err != nil {
+		// create a real handle to iptables
+		ipt, iptErr := iptables.New()
+		if iptErr != nil {
+			return fmt.Errorf("failed to detect iptables: %w", iptErr)
+		}
+		// most likely the pause container was removed from underneath nomad
+		return c.forceCleanup(ipt, alloc.ID)
+	}
+
+	return nil
+}
+
+// IPTables is a subset of iptables.IPTables
+type IPTables interface {
+	List(table, chain string) ([]string, error)
+	Delete(table, chain string, rule ...string) error
+	ClearAndDeleteChain(table, chain string) error
+}
+
+var (
+	// ipRuleRe is used to parse a postrouting iptables rule created by nomad, e.g.
+	//   -A POSTROUTING -s 172.26.64.191/32 -m comment --comment "name: \"nomad\" id: \"6b235529-8111-4bbe-520b-d639b1d2a94e\"" -j CNI-50e58ea77dc52e0c731e3799
+	ipRuleRe = regexp.MustCompile(`-A POSTROUTING -s (\S+) -m comment --comment "name: \\"nomad\\" id: \\"([[:xdigit:]-]+)\\"" -j (CNI-[[:xdigit:]]+)`)
```

Comment:
My kingdom for an iptables library that parses the rules into a sensible struct instead of returning a list of strings! 😀

Reply:
:pray: